src/connectors/clear-all/connectClearAll.js

```diff
@@ -46,7 +46,8 @@ const clearAll = ({helper, clearAttributes, hasRefinements}) => () => {
 /**
  * Connects a rendering with the clearAll business logic.
  * @function connectClearAll
- * @param {function(ClearAllRenderingOptions)} renderFn function that renders the clear all widget
+ * @param {function(ClearAllRenderingOptions, boolean)} renderFn function that renders the clear all widget,
+ *        if the boolean is true, then it's the rendering that happens before someone searches
```

Comment:
We might be able to provide a unique reference to a function that close a variable that we could update with the new values to clear each time there is a rendering. This way we could reference it in the first rendering and bind the correct event handler once.